scripts/protocgen.sh

```diff
@@ -1,23 +1,19 @@
-#!/usr/bin/env bash
+#!/bin/sh
 
 set -eo pipefail
 
-project_dir=x/wasm/internal/types/
-cosmos_sdk_dir=$(go list -f "{{ .Dir }}" -m github.com/cosmos/cosmos-sdk)
-# Generate Go types from protobuf
-protoc \
-  -I=. \
-  -I="$cosmos_sdk_dir/third_party/proto" \
-  -I="$cosmos_sdk_dir/proto" \
-  --gocosmos_out=Mgoogle/protobuf/any.proto=github.com/cosmos/cosmos-sdk/codec/types,plugins=interfacetype+grpc,paths=source_relative:. \
-  $(find "${project_dir}" -maxdepth 1 -name '*.proto')
+PROJECT_PROTO_DIR=x/wasm/internal/types/
+COSMOS_SDK_DIR=${COSMOS_SDK_DIR:-$(go list -f "{{ .Dir }}" -m github.com/cosmos/cosmos-sdk)}
 
-# Generate gRPC gateway (*.pb.gw.go in respective modules) files
+# Generate Go types from protobuf
 protoc \
   -I=. \
-  -I="$cosmos_sdk_dir/third_party/proto" \
-  -I="$cosmos_sdk_dir/proto" \
+  -I="$COSMOS_SDK_DIR/third_party/proto" \
+  -I="$COSMOS_SDK_DIR/proto" \
+  --gocosmos_out=Mgoogle/protobuf/any.proto=github.com/cosmos/cosmos-sdk/codec/types,Mgoogle/protobuf/empty.proto=github.com/gogo/protobuf/types,plugins=interfacetype+grpc,paths=source_relative:. \
   --grpc-gateway_out .\
   --grpc-gateway_opt logtostderr=true \
-  --grpc-gateway_opt paths=source_relative \
-  $(find "${project_dir}" -maxdepth 1 -name '*.proto')
+  --grpc-gateway_opt paths=Mgoogle/protobuf/any.proto=github.com/cosmos/cosmos-sdk/codec/types,Mgoogle/protobuf/empty.proto=github.com/gogo/protobuf/types,paths=source_relative \
+  --doc_out=./doc \
+  --doc_opt=markdown,proto.md \
+  $(find "${PROJECT_PROTO_DIR}" -maxdepth 1 -name '*.proto')
```

Comment:
This script is executed by the developer who changes the .proto files and the docs are checked-in along with the proto changes?

Reply:
yes, generated files are checked into the repo.